### PR TITLE
Allow saving as BigTIFF

### LINKED ISSUE #4669: Saving TIFF in chunks
### What did you do?
I need to save to save huge image files (approx. 819200x460800 RGBA). This is too much for anyone's RAM so I have to save it in chunks from disk. I start by saving the array to an HDF5 file. I then loop over the array in large steps, and parse a slice of the array into .fromarray(). I then save this to a tiff file. Once it loops again, it will add more to the tiff file and so on.

### What did you expect to happen?
It should create an tiff image that is very large.

### What actually happened?
It errored out while saving, giving me the error:
```pytb
ile "c:\Users\Dexter\Desktop\workspace\test.py", line 68, in test
    a.save("ff.tiff", format="tiff", quality=1)
  File "C:\Users\Dexter\AppData\Local\Programs\Python\Python38\lib\site-packages\PIL\Image.py", line 2134, in save
    save_handler(self, fp, filename)
  File "C:\Users\Dexter\AppData\Local\Programs\Python\Python38\lib\site-packages\PIL\TiffImagePlugin.py", line 1629, in _save
    offset = ifd.save(fp)
  File "C:\Users\Dexter\AppData\Local\Programs\Python\Python38\lib\site-packages\PIL\TiffImagePlugin.py", line 865, in save
    result = self.tobytes(offset)
  File "C:\Users\Dexter\AppData\Local\Programs\Python\Python38\lib\site-packages\PIL\TiffImagePlugin.py", line 808, in tobytes
    data = self._write_dispatch[typ](self, *values)
  File "C:\Users\Dexter\AppData\Local\Programs\Python\Python38\lib\site-packages\PIL\TiffImagePlugin.py", line 644, in <lambda>
    b"".join(self._pack(fmt, value) for value in values)
  File "C:\Users\Dexter\AppData\Local\Programs\Python\Python38\lib\site-packages\PIL\TiffImagePlugin.py", line 644, in <genexpr>
    b"".join(self._pack(fmt, value) for value in values)
  File "C:\Users\Dexter\AppData\Local\Programs\Python\Python38\lib\site-packages\PIL\TiffImagePlugin.py", line 611, in _pack
    return struct.pack(self._endian + fmt, *values)
struct.error: argument out of range
```
### What are your OS, Python and Pillow versions?

* OS:  Windows 10 V. 1909
* Python:  3.8.1
* Pillow:  Latest (7.1.2)

```python
def test():
    f = h5py.File("test.hdf5", "w")
    dset = f.create_dataset("test", (100000,100000,4), dtype=np.uint8, compression='gzip')

    shp = dset.shape
    step = 25000

    for i in range(step, shp[0]+step, step):
        a = Image.fromarray(dset[:i])
        a.save("out.tiff", format="tiff", quality=80) #should error out here
        del a
        gc.collect()

    f.close()
test()
```
I mention the size 819200x460800 - that's the maximum possible size. I also get the error on the size shown above.
If the size of the image is lower (for instance, 10000x10000) with a step size of 1000, it will not error and will produce an output image in about 4 seconds.

### PR DESCRIPTION
**Pull request by radarhere**
Resolves #4669

Adds a keyword argument to allow TIFF images to be saved as BigTIFF - `im.save("out.tiff", big_tiff=True)`

### CODE REVIEW

File: src/PIL/TiffImagePlugin.py [hunk 1]
```diff
@@ -582,7 +582,7 @@ class ImageFileDirectory_v2(_IFDv2Base):
 
     def __init__(
         self,
-        ifh: bytes = b"II\052\0\0\0\0\0",
+        ifh: bytes = b"II\x2A\x00\x00\x00\x00\x00",
```
Comment (radarhere) [inline src/PIL/TiffImagePlugin.py:585]: This is the same, I'm just changing it to the form seen elsewhere in this file.

File: src/PIL/TiffImagePlugin.py [hunk 2]
```diff
@@ -949,28 +949,34 @@ def load(self, fp: IO[bytes]) -> None:
             warnings.warn(str(msg))
             return
 
+    def _get_ifh(self):
+        ifh = self._prefix + self._pack("H", 43 if self._bigtiff else 42)
+        if self._bigtiff:
+            ifh += self._pack("HH", 8, 0)
+        ifh += self._pack("Q", 16) if self._bigtiff else self._pack("L", 8)
```
Comment (radarhere) [inline src/PIL/TiffImagePlugin.py:956]: See https://web.archive.org/web/20240706160214/https://www.awaresystems.be/imaging/tiff/bigtiff.html for a description of the header values.

File: docs/handbook/image-file-formats.rst [hunk 3]
```diff
@@ -1208,6 +1208,9 @@ The :py:meth:`~PIL.Image.Image.save` method can take the following keyword argum
 
     .. versionadded:: 8.4.0
 
+**bigtiff**
```
Comment (hugovk) [inline docs/handbook/image-file-formats.rst:1211]: I suggest adding an underscore to the public API for readability (and in the code):
```suggestion
**big_tiff**
```

Comment (radarhere) [reply docs/handbook/image-file-formats.rst]: Ok, done.

**Pull request merged**